src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2807,13 +2807,25 @@ class S4Importer final : public IParkImporter
                             ConvertWall(&type, &colourA, &colourB);
 
                             type = _wallTypeToEntryMap[type];
-                            auto edgeSlope = LandSlopeToWallSlope[slope][edge & 3] & ~EDGE_SLOPE_ELEVATED;
+                            auto baseZ = originalTileElement.GetBaseZ();
+                            auto clearanceZ = originalTileElement.GetClearanceZ();
+                            auto edgeSlope = LandSlopeToWallSlope[slope][edge & 3];
+                            if (edgeSlope & EDGE_SLOPE_UPWARDS || edgeSlope & EDGE_SLOPE_DOWNWARDS)
+                            {
+                                clearanceZ += LAND_HEIGHT_STEP;
+                            }
+                            if (edgeSlope & EDGE_SLOPE_ELEVATED)
+                            {
+                                edgeSlope &= ~EDGE_SLOPE_ELEVATED;
+                                baseZ += LAND_HEIGHT_STEP;
+                                clearanceZ += LAND_HEIGHT_STEP;
```

Comment:
This is equal in both `if`s, which hints like it could be extracted, unless there are vlaues other than the three `EDGE_SLOPE_*`s you mentioned.

Reply:
I don't know what you're talking about. It can one of the two branches, but also both.

Reply:
Ok, that also answers the other issue